File: java/collatz-conjecture/_iterations/1/src/main/java/CollatzCalculator.java
class CollatzCalculator {
    int computeStepCount(int start) {
        if (start <= 0) {
            throw new IllegalArgumentException("Only natural numbers are allowed");
        }
        
        int steps = 0;
        int number = start;
        while (number != 1) {
            number = (number % 2 == 0) ? number / 2 : 3 * number + 1;
            ++steps;
        }
        return steps;
    }
}

File: kotlin/collatz-conjecture/src/main/kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(num: Int): Int {
        require(num >= 1) { "Only natural numbers are allowed" }
        return generateSequence(num) { n -> if (n % 2 == 0) n / 2 else n * 3 + 1 }.indexOf(1)
    }
}
